lib/linter/linter.js

```diff
@@ -444,7 +440,10 @@ function normalizeEcmaVersion(ecmaVersion) {
     return ecmaVersion >= 2015 ? ecmaVersion - 2009 : ecmaVersion;
 }
 
-const eslintEnvPattern = /\/\*\s*eslint-env\s(.+?)\*\//gsu;
+const eslintEnvPatterns = [
+    /\/\*\s*eslint-env\s(.+?)\*\//gsu, // block comments
+    /\/\/\s*eslint-env\s(.+)/gu // line comments
```

Comment:
Some edge cases where `\s` consumes linebreaks:

```js
/* eslint no-undef: error */

// eslint-env
browser

window; // <- should be reported by no-undef

```

```js
/* eslint no-undef: error */

//
eslint-env
browser

window; // <- should be reported by no-undef
```

Reply:
```suggestion
    /\/\/[^\S\r\n\u2028\u2029]*eslint-env[^\S\r\n\u2028\u2029](.+)/gu // line comments
```

`[^\S\r\n\u2028\u2029]` = whitespace but not linebreak

Reply:
👍

Reply:
also found it in this line: https://github.com/eslint/eslint/blob/e2bed2ead22b575d55ccaeed94eecd3a979dd871/lib/rules/utils/ast-utils.js#L35